renderdoc/driver/metal/metal_types.cpp

```diff
@@ -121,4 +123,499 @@ void DoSerialise(SerialiserType &ser, NS::String *&el)
   }
 }
 
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, MTL::TextureSwizzleChannels &el)
+{
+  SERIALISE_MEMBER(red);
+  SERIALISE_MEMBER(green);
+  SERIALISE_MEMBER(blue);
+  SERIALISE_MEMBER(alpha);
+}
```

Comment:
By convention `DoSerialise` functions should all be in a `metal_serialise.cpp` file, similar to `metal_stringise.cpp`. At least for public-facing serialisation of API structs and objects.